econml/dml.py

```diff
@@ -485,8 +548,76 @@ def fit(self, Y, T, X=None, W=None, *, sample_weight=None, sample_var=None, grou
         self
         """
         return super().fit(Y, T, X=X, W=W, sample_weight=sample_weight, sample_var=sample_var, groups=groups,
+                           cache_values=cache_values, monte_carlo_iterations=monte_carlo_iterations,
                            inference=inference)
 
+    @property
+    def linear_first_stages(self):
+        return self._linear_first_stages
+
+    @linear_first_stages.setter
+    def linear_first_stages(self, linear_first_stages):
+        self._linear_first_stages = linear_first_stages
+        self._update_models()
+        # updating the models will set a misleading invalidation message, so overwrite it
+        self._cache_invalid_message = "Changing linear_first_stages invalidates stored nuisance results"
+
+    @_BaseDML.featurizer.setter
+    def featurizer(self, featurizer):
+        self._featurizer = clone(featurizer, safe=False)
+        if self._linear_first_stages:
+            self._update_models()
+            # updating the models will set a misleading invalidation message, so overwrite it
+            self._cache_invalid_message = ("Changing the featurizer when linear_first_stages is True "
+                                           "invalidates stored nuisance results")
+        else:
+            # only the final model needs to change
+            self._set_final_model()
+
+    @property
+    def fit_cate_intercept(self):
+        return self._fit_cate_intercept
+
+    @fit_cate_intercept.setter
+    def fit_cate_intercept(self, fit_cate_intercept):
+        self._fit_cate_intercept = fit_cate_intercept
+        # only the final model needs to change
+        self._set_final_model()
+
+    @property
+    def bias_part_of_coef(self):
+        return self.fit_cate_intercept
+
+    @_OrthoLearner.discrete_treatment.setter
+    def discrete_treatement(self, discrete_treatement):
+        # super().discrete_treatment = discrete_treatment
```

Comment:
In general: we shoulld change the name of the _RLearner.model_final to _RLearner.rlearner_model_final, and _OrthoLearner.model_final to _OrthoLearner.ortholearner_model_final and then we should just be setting these variables. That would clean up the code even if the refit stuff were not there.

Reply:
Agreed.